packages/network-controller/src/NetworkController.ts

```diff
@@ -43,16 +43,35 @@ export type NetworkDetails = {
 };
 
 /**
- * @type NetworkState
+ * Usually, you can use the provider object that the network controller exposes
+ * to talk to a network. However, the controller may be in a preliminary state,
+ * where no such provider exists, or an intermediate state, where a provider
+ * exists but the network is not accessible. This enum represents the positive
+ * and negative states.
  *
+ * @property active - Represents the state in which a provider object has been
+ * created for the currently selected network and the network is provably
+ * accessible.
+ * @property loading - Represents the state in which a provider object does not
+ * exist for the currently selected network, or it does exist but the network is
+ * provably inaccessible.
```

Comment:
Nit: Maybe better to use softer language here since we only check periodically, and since the way we check today is a bit questionable and might not be a guarantee that the network is accessible (e.g. if a network doesn't implement `net_version`, or uses a cache for static methods like this, it might not reflect whether the network is accessible or not).
```suggestion
 * created for the currently selected network and the network is seems
 * accessible.
 * @property loading - Represents the state in which a provider object does not
 * exist for the currently selected network, or it does exist but the network is
 * seems inaccessible.
```